src/main/java/com/elixr/poc/service/PurchaseService.java

```diff
@@ -0,0 +1,44 @@
+package com.elixr.poc.service;
+import com.elixr.poc.data.Purchase;
+import com.elixr.poc.repository.PurchaseRepository;
+import com.elixr.poc.rest.request.PurchaseRequest;
+import com.elixr.poc.rest.response.PurchaseResponse;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+/**
+ * service class for editing response and save or update new purchase to database
+ */
+
+@Service
+public class PurchaseService {
+    private final PurchaseRepository purchaseRepository;
+
+    public PurchaseService(PurchaseRepository purchaseRepository) {
+        this.purchaseRepository = purchaseRepository;
+    }
+
+    private static Purchase createPurchaseObjectFromRequest(PurchaseRequest purchaseRequest) {
```

Comment:
Service should not know anything about rest objects. 
createPurchaseObjectFromRequest() should be done inside controller and purchase object should be passed

Reply:
Updated